[#10433] Instructor view results: Use fixed-width floating numbers

swood049 opened the pull request:
Fixes #10433

I added a DecimalPipe object inside of mcq-question-statistics.componenets.ts, and I used it to format the information given to the table as requested in the issue (with 3 significant digits). It changes only the Average and Total columns, as well as the weight numbers in the header.

madanalogy commented on e:
```diff
@@ -0,0 +1,388 @@
+[33mcommit 1c2b94d63a7be533d96e149223c0f4368bcea1cd[m[33m ([m[1;36mHEAD -> [m[1;32mmaster[m[33m, [m[1;31mupstream/master[m[33m)[m
```
What is this file? Could you remove it from your PR?

swood049 replied:
I believe that this is referring to the module "DecimalPipe" module that I am using to make the change that is asked for. I am not 100% certain on how to use the module without importing it.

swood049 replied:
Upon further review, I have no idea what this file is, and I do not think that I made that change.

swood049 replied:
I made the changes requested.

madanalogy reviewed (changes requested):
Your PR seems to be failing the lint check, please execute `npm run lint` and fix the errors that pop up

madanalogy commented:
Could you please add some screenshots as well to verify the fix? Thanks!

swood049 reviewed (commented):
I made the requested changes.

madanalogy commented:
> I made the requested changes.

Please see my comment above regarding the addition of screenshots

swood049 commented:
The first photo below shows that I deleted the file, and that I moved the location of the import statement (it also says the lint test passes). 
![image](https://user-images.githubusercontent.com/38500060/100440280-a6eeb980-3072-11eb-9760-ebfdd04d2b39.png)
I just tested it in my local development. Everything seems to be working as requested.
Proof that I made the change:
![image](https://user-images.githubusercontent.com/38500060/100442547-33e74200-3076-11eb-951e-57a6efdf2c86.png)

The local lint test: 
![image](https://user-images.githubusercontent.com/38500060/100442472-1619dd00-3076-11eb-887b-baf36c8c65e8.png)

Check to see that it runs:
![image](https://user-images.githubusercontent.com/38500060/100442614-49f50280-3076-11eb-8b60-7025b3b00387.png)

madanalogy commented:
> I think that these are what you need. If I am wrong, let me know.

No that is not what I meant sorry. Refer to the original issue filed, there are screenshots showing the numbers compared between v6 and v7 https://github.com/TEAMMATES/teammates/issues/10433#issue-665729617

What you need to do is replicate the process (i.e. create a feedback session, then submit dummy statistics) and take a screenshot showing your changes. This screenshot should show the same UI component as the original issue filed.

swood049 commented:
Ah! Yes! Got it!
I took the liberty to do the same with the column titles.
![image](https://user-images.githubusercontent.com/38500060/100495520-9003a200-311a-11eb-811f-3a984de0da7b.png)

madanalogy commented on src/web/app/components/question-types/question-statistics/mcq-question-statistics.component.ts:
```diff
@@ -62,7 +66,7 @@ export class McqQuestionStatisticsComponent extends McqQuestionStatisticsCalcula
       ...Object.keys(this.weightPerOption).map((key: string) => {
         return {
-          header: `${key} [${this.weightPerOption[key]}]`,
+          header: `${key} [${this.changeSignificantDigits.transform(this.weightPerOption[key], '1.2-3')}]`,
```
I'm not keen on the use of this string variable in-line, feels too much like a magic number. Could you maybe define a constant variable with a descriptive name? Preferably in the scope of this method.

madanalogy commented:
Closing due to inactivity
